mods/default/nodes.lua

```diff
@@ -586,29 +586,124 @@ minetest.register_node("default:snow", {
 	inventory_image = "default_snowball.png",
 	wield_image = "default_snowball.png",
 	paramtype = "light",
+	paramtype2 = "leveled",
 	buildable_to = true,
 	floodable = true,
+	sunlight_propagates = true,
+	walkable = true,
 	drawtype = "nodebox",
+	leveled = 7,
 	node_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -0.25, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	collision_box = {
-		type = "fixed",
-		fixed = {
-			{-0.5, -0.5, -0.5, 0.5, -7 / 16, 0.5},
-		},
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
+	},
+	selection_box = {
+		type = "leveled",
+		fixed = {-0.5, -0.5, -0.5, 0.5, -0.5, 0.5},
 	},
 	groups = {crumbly = 3, falling_node = 1, snowy = 1},
 	sounds = default.node_sound_snow_defaults(),
 
 	on_construct = function(pos)
-		pos.y = pos.y - 1
-		if minetest.get_node(pos).name == "default:dirt_with_grass" then
-			minetest.set_node(pos, {name = "default:dirt_with_snow"})
+		pos.y = pos.y-1
+		local node = minetest.get_node(pos)
+		if node.name == "default:dirt_with_grass"
+		or node.name == "default:dirt" then
```

Comment:
What is about savanna dirt, rainforest litter, dirt with grass and footstep and co.?

Reply:
It was originally only:
```
if minetest.get_node(pos).name == "default:dirt_with_grass" then
    minetest.set_node(pos, {name = "default:dirt_with_snow"})
```

Reply:
I know, but as you are changing this, it might make sense to support all types of dirt.

Reply:
OK I'll do that.

Reply:
OK I'll do that. Any in specific, i.e which ones do I leave out? Or just add them all?
```
default:dirt
default:dirt_with_grass
default:dirt_with_grass_footsteps
default:dirt_with_dry_grass
default:dirt_with_snow
default:dirt_with_rainforest_litter
default:dirt_with_coniferous_litter

default:permafrost
default:permafrost_with_stones
default:permafrost_with_moss

default:sand
default:desert_sand
default:silver_sand
```

Reply:
OK I'll do that. Any in specific, i.e which ones do I leave out? Or just add them all?
```
default:dirt
default:dirt_with_grass
default:dirt_with_grass_footsteps
default:dirt_with_dry_grass
default:dirt_with_snow
default:dirt_with_rainforest_litter
default:dirt_with_coniferous_litter

default:permafrost
default:permafrost_with_stones
default:permafrost_with_moss

default:sand
default:desert_sand
default:silver_sand
```

Reply:
Well, sand becoming dirt_with_snow doesn't really make sense. Permafrost doesn't either.

~~All the other nodes have group `soil = 1`. Maybe use this. However, there might be non-dirt nodes that use the `soil` group. Sadly it isn't documented. (Actually farming seems to override desert_sand to make it a soil.)~~ No, the `soil` group can't be used.

Just make all the dirty nodes snowy (and ignore `dirt_with_snow`) and maybe store them in a table, idk.

Reply:
Yes just the 'dirt_with' nodes except '... footsteps' as that is a very rare node only found in old worlds.

Reply:
@DS-Minetest Why ignore `dirt_with_snow`?

Reply:
Well, `dirt_with_snow` is already `dirt_with_snow`, so you don't have to replace it with `dirt_with_snow`.

Reply:
Oh yeah, of course xD I didn't see that. I think that's already done.